social.md

```diff
@@ -76,7 +76,7 @@ The library must implement provider callbacks for all social provider
 directories that are mapped to the specified Stormpath application.  The
 callback URLs for each provider should take the form of:
 
-> `/<stormpath.web.socialProviders.callbackRoot>/<providerId>`
+> `<stormpath.web.socialProviders.[providerId].callbackUri>/<providerId>`
```

Comment:
I believe we should maintain `callbackUri` as it describes what this URI is used for.  Without it, one might think that this URI is used to initiate the OAuth workflow.

We need different callback URIs to distinguish which provider flow is occurring.  When we move the social OAuth workflows to the REST API, the provider context will be known because the callback URI is going to hang off a directory, and the directory has the provider context.  In the framework case, we have no idea what’s going on because the callback parameters (access token or code) is opaque.

@timothyej no, the dot should not be removed.  I want separate hashes for each provider.

I do like @lhazlewood 's proposal of  `socialProviers` -> `social`, so I will update this PR with that change.